.github/workflows/ci.yml

```diff
@@ -102,10 +103,7 @@ jobs:
         uses: actions/cache@v3
         with:
           path: .next/cache
-          key: ${{ runner.os }}-next-cache-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-next-cache-${{ github.sha }}
-            ${{ runner.os }}-next-cache-
+          key: ${{ runner.os }}-next-cache-${{ hashFiles('./package-lock.json', './next.config.js', './rewrites.js') }}
```

Comment:
What's the rationale about this? What does it improve over previous version?

Reply:
I think I was troubleshooting why the .next/cache was never being used and attributed it to the previous cache key. so I was testing with other key values. This turned out not the be the case and the issue was the change at `build_next.sh`.

Other than that I think we might want to use the package lock as part of the cache key to prevent using a cache built with a different module version. https://nextjs.org/docs/advanced-features/ci-build-caching#github-actions

Reply:
Fyi,

- `${{ runner.os }}-next-cache-${{ github.sha }}` was restoring the cache from the exact same commit
- `${{ runner.os }}-next-cache-` was restoring whatever cache we have -> it's ok to have an older staled cache, next.js is gonna use it and avoid recompiling anything it finds there

I'm ok to change the cache key, but I think it's good to keep this approach of partial keys. 

A cache is not a build, the compiler will figure if there are some things that are not compatible in there.

Reply:
I see. I was thinking for this step there would never be a sha cache ready unless we are retrying the entire workflow. It would be useful for the e2e build that happens next though.

In the next docs they include a hash of the package lock file in the cache key, would next be smart to invalidade the cache if some source files are unchanged but we have updated their node_modules dependencies?

Reply:
> would next be smart to invalidade the cache if some source files are unchanged but we have updated their node_modules dependencies

I do expect this. Think about your development environment, it uses the the .next cache heavily and you never have to flush it, it's smart enough to rebuild whatever was updated.